packages/gator-permissions-controller/src/types.ts

```diff
@@ -245,6 +253,11 @@ export type RevocationParams = {
    * The permission context as a hex string that identifies the permission to revoke.
    */
   permissionContext: Hex;
+
+  /**
+   * The metadata associated with the permission revocation transaction.
+   */
+  metadata?: RevocationMetadata;
```

Comment:
Shouldn't this be required? Is there a scenario where we want to persist the revocation status, but don't have revocation metadata?

As per other comments, I think we should name this explicitly `revocationMetadata` to make it clear to what it is pertaining.